globus_sdk/local_endpoint/personal.py

```diff
@@ -49,7 +51,7 @@ def endpoint_id(self):
                 if _on_windows():
                     appdata = os.getenv("LOCALAPPDATA")
                     if appdata is None:
-                        raise ValueError(
+                        raise GlobusSDKUsageError(
```

Comment:
Is the absence of `LOCALAPPDATA` on a Windows system the fault of usage, or is it the fault of a broken Windows installation? I don't think it matters, and we should probably leave it alone as it would be a breaking change, but maybe an `OSError` would have been better here.

Reply:
I was considering it a usage error from the start because it's not okay to try to use this method after clearing `os.environ` or invoking python with a really messed up env in the first place.
It's not possible for us to know why it wasn't set. So I just assume that it's the caller's fault at some level! :smile: 

I doubt this will ever get tripped, but #robustnessprinciple

Anyway, it sounds like you're okay with this error as it is in this PR.